torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -587,9 +590,24 @@ struct GraphFuser {
     auto* g = inputs[0]->owningGraph();
     auto* input_list =
         g->insertNode(g->createList(TensorType::get(), inputs))->output();
+    aliasDb_->createValue(input_list);
     auto* output_list = g->insert(aten::broadcast_tensors, {input_list});
+    aliasDb_->createValue(output_list);
     auto* unpack_node = g->insertNode(
         g->create(prim::ListUnpack, {output_list}, inputs.size()));
+
+    // We are doing:
+    //   input_list = listConstruct(a, b, ...)
+    //   output_list = broadcast_tensors(input_list)
+    //   a_broadcasted, b_broadcasted = listUnpack(output_list)
+    // `a_broadcasted` should receive the same aliasing info as `a`
+    TORCH_INTERNAL_ASSERT(unpack_node->outputs().size() == inputs.size());
```

Comment:
Technically `a_broadcasted` may or may not alias `a` but  i guess it is conservative to say that they alias

Reply:
yeah, broadcasting tensors may create an alias so similar to other ops (contiguous, etc.) we assume that it will